OOP/OOP-3/TrainReservation.java

```diff
@@ -0,0 +1,322 @@
+	Payment(String[] temp)
+	{
+		this.temp=temp;
+	}
+	public ArrayList<String> paymentType()
+	{
+		int i=0;
+		ArrayList<String> bookingInformation=new ArrayList<String>();
+		System.out.println("\n1:CC\n2:Wallet\n3:Net Banking");
+		int hold=sc.nextInt();
+		switch(hold)
+		{
+		case 1:System.out.println("\n Enter  CC number");
+		String cc=sc.next();
+		bookingInformation.add(cc);
+		System.out.println("\n Enter cvv number");
+		String cvv=sc.next();
+		bookingInformation.add(cvv);
+		System.out.println("Successful Booking");
+		break;
+		case 2:System.out.println("Enter amount:");
+		int amount=sc.nextInt();
+		bookingInformation.add(Integer.toString(amount));
+		System.out.println("Successfull Booking");
+		break;
+		case 3:System.out.println("Enter Bank Name");
+		String bankName=sc.next();
+		bookingInformation.add(bankName);
+		System.out.println("Enter user name:");
+		String userName=sc.next();
+		bookingInformation.add(userName);
+		System.out.println("Enter password:");
+		String password=sc.next();
+		bookingInformation.add(password);
+		System.out.println("Successfull Booking");
+		}
+		return bookingInformation;
+	}
+}
+//this class is used to hold personal information of user 
+class PersonalInformation
+{
+	private String name;
+	public PersonalInformation(String name)
+	{
+		this.name=name;
+	}
+	public void setName(String name)
+	{
+		this.name=name;
+	}
+	public String getName()
+	{
+		return name;
+	}
+}
+class TrainBooking
+{
+	Scanner sc;
+	int flag=0;
+	ArrayList<String> paymentInformation;
+	String trainNumber,from,to,trainType,name;
+	int numberOfSeats;
+	ArrayList<String> trainInfo;
+	TrainBooking(String trainNumber,int numberOfSeats,ArrayList<String> trainInfo,String from,String to,String trainType,String name)
+	{
+		this.trainNumber=trainNumber;
+		this.numberOfSeats=numberOfSeats;
+		this.trainInfo=trainInfo;
+		this.from=from;
+		this.to=to;
+		this.trainType=trainType;
+		this.name=name;
+	}
+	//this method is used for printing the booking information
+	public void booking()
+	{
+		try
+		{
+			for(int i=0;i<trainInfo.size();i++)
+			{
+				String temp[]=trainInfo.get(i).split(",");
+				if(temp[0].equalsIgnoreCase(trainNumber) && Integer.parseInt(temp[3])>=numberOfSeats)
+				{
+					PersonalInformation pInformation=new PersonalInformation(name);
+					System.out.print("Payment Detail");
+					Payment payment=new Payment(temp);
+					paymentInformation=payment.paymentType();
+					System.out.println("Deatail:"+name+paymentInformation);
+					flag++;
+					TrainChart trainChart=new TrainChart(trainType, trainInfo);
+					trainChart.updatedChart(numberOfSeats,trainNumber);
+				}
+			}
+		}catch(Exception e){}
+		if(flag==0)
+		{
+			System.out.println("sorry no seats available");
+		}
+		//return paymentInformation;
+	}
+}
+//main class for train reservation
+public class TrainReservation  
+{
+	public static void main(String args[])throws FileNotFoundException,NullPointerException,IOException
```

Comment:
If you write exceptions in throws block, It will throw these exception to JVM and then JVM will show error stack trace. Instead of writing exceptions in throws block, handle them in try catch block.
Please follow these links for better understanding of exception handling.

http://javarevisited.blogspot.in/2012/02/difference-between-throw-and-throws-in.html
http://javarevisited.blogspot.in/2013/03/0-exception-handling-best-practices-in-Java-Programming.html
http://javarevisited.blogspot.in/2013/06/10-java-exception-and-error-interview-questions-answers-programming.html